uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/AbsenceType.php

```diff
@@ -28,4 +29,62 @@ private function onToilDisable(AbsenceType $absenceType) {
     $startDate = new DateTime($currentPeriod->start_date);
     LeaveRequest::deleteAllNonExpiredTOILRequestsForAbsenceType($absenceType->id, $startDate);
   }
+
+  /**
+   * Checks if an absenceType has ever been used for a leave request
+   *
+   * @param int $absenceTypeID
+   *
+   * @return boolean
+   */
+  public function absenceTypeHasEverBeenUsed($absenceTypeID) {
+
+    if ($this->absenceTypeIsLinkedToLeaveRequest($absenceTypeID)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Deletes the AbsenceType with the given id.
+   * Checks first to see if the absence type can be deleted or not.
+   *
+   * @param int $absenceTypeID
+   *
+   * @throws CRM_HRLeaveAndAbsences_Exception_OperationNotAllowedException
+   * @throws UnexpectedValueException
+   */
+  public function delete($absenceTypeID) {
+    $absenceType = AbsenceType::findById($absenceTypeID);
+
+    if ($this->absenceTypeHasEverBeenUsed($absenceType->id)) {
+      throw new UnexpectedValueException('Absence type cannot be deleted because it is linked to one or more leave requests');
```

Comment:
I think it would make more sense to return the same exception (OperationNotAllowed) for both, after all they are just 2 different scenarios where it's not allow to delete an absence type